perfkitbenchmarker/benchmark_spec.py

```diff
@@ -228,6 +233,34 @@ def ConstructVirtualMachines(self):
       self.vm_groups[group_name] = vms
       self.vms.extend(vms)
 
+  def ConstructSparkService(self):
+    if self.config.spark_service is None:
+      return
+
+    providers.LoadProvider(self.config.spark_service.cloud)
+    spark_service_spec = self.config.spark_service
+    service_type = spark_service_spec.spark_service_type
+    if service_type == spark_service.PKB_MANAGED:
+      spark_service_class = spark_service.GetSparkService(
+          spark_service.PKB_MANAGED)
+    elif service_type == spark_service.PROVIDER_MANAGED:
+      spark_service_class = spark_service.GetSparkService(
+          spark_service_spec.cloud)
+    else:
+      valid_types = ' '.join(spark_service._SPARK_SERVICE_REGISTRY.keys())
```

Comment:
Also I think this kind of validation of spark_service_type is meant to be done inside the spec decoder